internal/mempool/reactor.go

```diff
@@ -58,34 +56,27 @@ type Reactor struct {
 
 // NewReactor returns a reference to a new reactor.
 func NewReactor(
-	ctx context.Context,
 	logger log.Logger,
 	cfg *config.MempoolConfig,
-	peerMgr PeerManager,
 	txmp *TxMempool,
 	chCreator p2p.ChannelCreator,
-	peerUpdates *p2p.PeerUpdates,
-) (*Reactor, error) {
-
-	ch, err := chCreator(ctx, getChannelDescriptor(cfg))
-	if err != nil {
-		return nil, err
-	}
-
+	peerEvents p2p.PeerEventSubscriber,
+	getPeerHeight PeerHeightFetcher,
```

Comment:
I feel like this case would be fine with the function type, it's short, and saves an indirection to figure out what it really is
```suggestion
	getPeerHeight func(types.NodeID) int64,
```